cellprofiler/pipeline.py

```diff
@@ -1287,6 +1287,23 @@ def savetxt(self, fd_or_filename,
             write_file_list(fd, self.__file_list)
         if needs_close:
             fd.close()
+    
+    def save_pipeline_notes(self, fd, indent = 2):
```

Comment:
I've often written one of these intending to just write to a file, then later found that I had to embed the output as a section in another file. It's more flexible to leave the decision about whether to close to the caller - for instance, you might use this as part of a web page. And you can see the clumsiness of it in line 1289 - unfortunate logic which resulted from savetxt being called from different places with different requirements for the file's handling.

It also makes testing easier if you can write to StringIO.